Домашняя страница с афишей

ilyainavoid opened the pull request.

ilyainavoid committed:
feat(ui): add text and title components

ilyainavoid committed:
feat(routing): add router and specify a homepage path

ilyainavoid committed:
feat(layout): create header component

ilyainavoid committed:
fix: change "key" prop to "route"

ilyainavoid committed:
feat(api): install axios and create axios instance

Create axios instance and configure it:
1. Add base url and content type
2. Add interceptors to log requests and responses

ilyainavoid committed:
feat(api): get today's films request

Add function that fetches today's films and interfaces that describe some of the response's fields

ilyainavoid committed:
feat: add fetchTodaysFilms hook

ilyainavoid committed:
fix: add missed genres field into Film interface

ilyainavoid committed:
fix: add prop for classname in text and title

ilyainavoid committed:
fix: add missed user ratings field

ilyainavoid committed:
feat: add button with different types

ilyainavoid committed:
feat(ui): add extra text style

ilyainavoid committed:
feat(ui): add fractional rating component

ilyainavoid committed:
feat: create playbill page

Create film card to display brief film data, add blank film page to check redirecting on button click

danax7 commented on src/components/FilmCard/FilmCard.tsx:
```diff
@@ -0,0 +1,47 @@
+            <div className={styles.descriptionContainer}>
+                <div>
+                    <Title level={2} style={"title-bold"}>{film.name}</Title>
+                    <Text style={"text-regular"} className='translucent'>{film.originalName}</Text>
+                </div>
+                <div>
+                    <FractionalRating rating={Number(film.userRatings.kinopoisk)}/>
+                    <Text style={"text-regular"} className='translucent' fractions={5} step={2}>Kinopoisk - {film.userRatings.kinopoisk}</Text>
```
Строковые значения `<Text style={"text-regular"}` можно не оборачивать в фигурные скобки

kvartycs commented on src/components/FilmCard/FilmCard.tsx:
```diff
@@ -0,0 +1,47 @@
+import {FC} from "react";
+import {Film} from "../../interfaces/film.ts";
+import styles from './filmcard.module.scss'
+import {BASE_URL} from "../../consts/baseURL.ts";
+import FilmInfoLabel from "../GenreAndCountryLabel/FilmInfoLabel.tsx";
+import Title from "../UI/Title/Title.tsx";
+import Text from '../UI/Text/Text.tsx'
+import FractionalRating from "../UI/FractionalRating/FractionalRating.tsx";
+import Button from "../UI/Button/Button.tsx";
+import {Link, useNavigate} from "react-router-dom";
+import {ROUTES} from "../../consts/routes.ts";
```
алиасы надо попробовать затащить тебе сюда, и линтер добавить, чтобы импорты тут автоматически правильно выстраивались

сначала библиотечные импорты, потом  компоненты и прочие вещи, а в конце уже стили

kvartycs commented on src/App.tsx:
```diff
@@ -1,9 +1,11 @@
+import {RouterProvider} from "react-router-dom";
 
 function App() {
 
   return (
-    <></>
+      <RouterProvider router={router}/>
   )
 }
```
у тебя везде по проекту стрелочные функции, тут тоже надо переделать

kvartycs commented on src/components/UI/Text/Text.tsx:
```diff
@@ -0,0 +1,16 @@
+import {FC, ReactNode} from "react";
+import '../../../styles/text.scss'
```
у тебя вроде компоненты одни сделаны правильно, а другие вот так, должно быть единообразие (не тащить не понятно откуда стили)

плюс в компонентах выше module.scss файлы тоже с большой буквы как и сам компонент (Button.module.scss)

kvartycs commented on src/components/UI/Text/Text.tsx:
```diff
@@ -0,0 +1,16 @@
+
+const Text: FC<TextProps> = ({children, style, className}) => {
+    return (
+        <p className={`text ${style} ${className}`}>{children}</p>
```
такие стили лучше через библиотеку classnames писать

kvartycs commented on src/components/UI/Title/Title.tsx:
```diff
@@ -0,0 +1,40 @@
+const Title: FC<TitleProps> = ({ children, level, style, className }) => {
+    let Tag: ElementType;
+
+    switch (level) {
+        case 1:
+            Tag = 'h1';
+            break;
+        case 2:
+            Tag = 'h2';
+            break;
+        case 3:
+            Tag = 'h3';
+            break;
+        case 4:
+            Tag = 'h4';
+            break;
+        case 5:
+            Tag = 'h5';
+            break;
+        case 6:
+            Tag = 'h6';
+            break;
+        default:
+            Tag = 'h1';
+    }
```
зачем такой велосипед, когда можно сразу прокидывать туда h1 | h2 | h3 ...

kvartycs commented on src/pages/PlaybillPage/PlaybillPage.tsx:
```diff
@@ -0,0 +1,24 @@
+import FilmCard from "../../components/FilmCard/FilmCard.tsx";
+
+const PlaybillPage: FC = () => {
+    const filmsList = useFetchTodaysFilms();
+
+    return (
+        <section className={styles.playbillContainer}>
+            <div className={styles.playbillContent}>
+                <Title level={1} style="text-bold" className={styles.playbillHeading}>Афиша</Title>
+                <div className={styles.postersContainer}>
+                    {filmsList && filmsList.map(film => (
+                        <FilmCard key={film.id} film={film} />
+                    ))}
+                </div>
+            </div>
+        </section>
+    )
+}
```
мне не очень нравится эта история с &&

лучше хук переделать чтобы он тебе loading возвращал еще, тут через деструктуризацию достать и потом писать типо loading && (Было бы идеально если бы ты тут лоадер сделал чтобы был лучше UX)

kvartycs commented on src/utils/api/axiosInstance.ts:
```diff
@@ -0,0 +1,33 @@
+
+axiosInstance.interceptors.request.use(
+    (config) => {
+        console.log(`Запрос: ${config.method?.toUpperCase()} ${config.url}`, config);
+        return config;
+    },
+    (error) => {
+        console.error('Ошибка запроса:', error);
+        return Promise.reject(error);
+    }
+);
+
+axiosInstance.interceptors.response.use(
+    (response) => {
+        console.log(`Ответ: ${response.status} ${response.statusText}`, response);
+        return response;
+    },
+    (error) => {
+        console.error('Ошибка:', error);
+        return Promise.reject(error);
+    }
+);
```
понимаю что логи для теста, но в будущем нужно их будет обязательно убрать

kvartycs commented on src/utils/api/cinema/today/todayRequestResponse.ts:
```diff
@@ -0,0 +1,7 @@
+import {Film} from "../../../../interfaces/film.ts";
```
еще одна причина затащить сюда алиасы

ilyainavoid committed:
refactor: edit text and title components

Re-write styles with classnames lib
Bring styles to the same dir
In Title edit the way of choosing a heading type

ilyainavoid committed:
refactor: edit app component

Make App component a function expression

ilyainavoid committed:
refactor: prettier config, format code

ilyainavoid committed:
chore: configure eslint, stylelint and prettier

ilyainavoid committed:
refactor: format files

ilyainavoid committed:
chore: add script fot eslint --fix

ilyainavoid committed:
chore: install and configure husky and lint-staged

ilyainavoid committed:
refactor: change module files names

ilyainavoid committed:
refactor: replace index.css with global and reset css files

ilyainavoid committed:
fix: prop name in header section

ilyainavoid committed:
refactor: fix line separators

Some issue occurred with line separators. Prettier highlighted all lines and suggested to fix it.

ilyainavoid committed:
chore: reconfigure prettier to fix end of line separator issue

ilyainavoid committed:
refactor: fix paths with aliases, edit ui elements props, edit styles of button

ilyainavoid committed:
feat: add loader and skeleton for unloaded image
